src/Classes/TreeTab.lua

```diff
@@ -156,22 +156,35 @@ local TreeTabClass = newClass("TreeTab", "ControlHost", function(self, build)
 		return self.showConvert
 	end
 	self.controls.specConvert = new("ButtonControl", {"LEFT",self.controls.specConvertText,"RIGHT"}, 8, 0, 120, 20, "^2Convert to "..treeVersions[latestTreeVersion].display, function()
-		local newSpec = new("PassiveSpec", self.build, latestTreeVersion)
-		newSpec.title = self.build.spec.title
-		newSpec.jewels = copyTable(self.build.spec.jewels)
-		newSpec.tree.legion.editedNodes = self.build.spec.tree.legion.editedNodes
-		newSpec:RestoreUndoState(self.build.spec:CreateUndoState())
-		newSpec:BuildClusterJewelGraphs()
-		t_insert(self.specList, self.activeSpec + 1, newSpec)
-		self:SetActiveSpec(self.activeSpec + 1)
-		self.modFlag = true
-		main:OpenMessagePopup("Tree Converted", "The tree has been converted to "..treeVersions[latestTreeVersion].display..".\nNote that some or all of the passives may have been de-allocated due to changes in the tree.\n\nYou can switch back to the old tree using the tree selector at the bottom left.")
+		self:ConvertSpec(self.build.spec)
 	end)
 	self.jumpToNode = false
 	self.jumpToX = 0
 	self.jumpToY = 0
 end)
 
+function TreeTabClass:ConvertSpec(spec, specIndex)
+	-- Parameters:
+		-- Mandatory : spec
+		-- Optional : specIndex - use when the the spec being converted is not the active one
+
+	local specIdx = specIndex or self.activeSpec
+	local newSpec = new("PassiveSpec", self.build, latestTreeVersion)
+
+	newSpec.title = spec.title
+	newSpec.jewels = copyTable(spec.jewels)
+	newSpec.tree.legion.editedNodes = spec.tree.legion.editedNodes
+	newSpec:RestoreUndoState(spec:CreateUndoState())
+	newSpec:BuildClusterJewelGraphs()
+	t_insert(self.specList, specIdx + 1, newSpec)
+	self.modFlag = true
+	if not specIndex then
+		-- only SetActiveSpec if we are converting from the main screen 
+		self:SetActiveSpec(specIdx + 1)
+		main:OpenMessagePopup("Tree Converted", "The tree has been converted to "..treeVersions[latestTreeVersion].display..".\nNote that some or all of the passives may have been de-allocated due to changes in the tree.\n\nYou can switch back to the old tree using the tree selector at the bottom left.")
```

Comment:
This doesn't work exactly how you might expect, here.  If you select the 3.15 default build, and convert it from the pop-up, it'll auto-switch to the build right above it: 
```bash
eNrtPWlz2kjan4dfoXLVvJVUAOs-_Ca7hTG2cSCxwUeSL1MyNKBYSFiHCdnyf9-nD0Anapjs1O7WTqYyGD331U8_3fK8__uPuSu8oCB0fO_DkdQUjwTkjfyx400_HN3dnjfMo7__rfb-2o5mnyensePiJ3-r_faefBZc9ILcD0cmoEV2MEXR_ZqU8gd8t7C9aIZ8r29_94MLf_zh6JPvoSPh0fbGTvThqOU6gX0kjFw7DD_Zc_Th6BbNF64dHAl2OELeuL19cumgwF_MgOKRMLcdb-iPnlB0EfjxAgQ_El4ctOz7Y4Ds9q8_D24T3B0vyR2k_-39tWuvUDCM7EgI4S8QBYxgT9GlA-RfbDcGOuLRcTHscIHQuBrsOkCdyQSNIucFtQMnaoPwI1SNty9sP3YjZ-GCfTbwUtPSylBu_ch2z66H1cQppM9hkQcnmp26YBUuuhi6O_WcCHGDX_tO6Ht7Sc0F3I5dFyKaD9afPzoep45t33fH_tJLuESWpTLoVoDszxMaLQN77MThBs9Sy5D6tme3_ZDDPT1ngvggO0M-uAHkCh8klvIaBZDNEb-weyEwDkM08qFi7MNjH5SEHocxOwCzMzwE4QBGw2hbO0yl1OnoOQmoiFYZ5Bn6sQFT1R30koCaUQbY9bY6KOou-ZKQsqyVy_fiR7BOcVaTzuX1No9VWdGahqSommgZeqlFifVh6XHmttu3fzjzeA6Ly639hLZcDcWwmroqm5oBZK1dAZTQSpd2LEyjEwzd9UYbBF3ZRffOC1CIgpfEgraLQxqFBdvWNuLOJBigaVJ5ualo8i74HkKj2QWs9QM7QnwZug0Tc4dvsJkwdNJMkqXsIlxgJ0WX1abFh5S1lKU0NVnSDcmUdb08TjGRtNVMSWyq5i74Pa32YAccvUzHQ8F0NZw5yN1Cq7ph7bRxEitla0XiYZTWnEu2PZXvvNhhsgxIorY7ahhCUpnyeETQEQHCGGW6Oam8Y_S_437R3Q-tFcz9OEjkr7xTBwqeVEEu72Fnq9AZQT9lz2HFH6BxPErVzdLavul8-_4LmkPok6YZGvREQ1S-Mpy60N1n1N-plesWoZSq1Yoie_R05o-nOSOrO7kcgDKMFwsoAzhseMU7dwKwdegklu6GygH9GTYxbXvB053i7MowqAbmpt9zprPIg8Y6w0TnROBXZGb7YVYTnQecm8U5bBG4Wv5rfwkkZ3gfHO4HDf1BdUadB8j7ueKmnwJPMlDK6-g4DnCAcvPIYuTZvD8mgwL8qTtf-EFEvmzb7igkJLveIo4Ej-zw5044-uMxnkzwNv4IWARkCtE5P--0b7v3HSZFEiV8clz3Dy-eP-LNL_3vtlAOESlWwsh3XXsRovGHo4nthkDbgY9DjDuEGj2KuOChe6NVkAsaltVR4IAReYDxHptLZDJ44IHEcwEuQPCK7SJ-g92uFghHQMhrM1bzeKDJBIELko4D-BQkgwY-N6CRveKC3LQZXNAdF7UcFy-CfFbrQyLQ9ZMPHlazwHmMI85oI9sJLjlwQ82lYLI75MwOPkDW2nAJwXo5Hli2YvAFJfQV3G47QxME4c6Z9qSc0C0hLZbrwvie5FoohFAxL9A8PF3BCnSOzZsZQ83tCAIRzXt4EHvr4zpsjyIU9OhgljFLkflwFAUxfDlGEzt28fc3se060QoX-sS3jAT0hkI485e4jaFkcAkIwTW9Hn3SciNGAfNYM6X6YzXIuLZFmkHyM1GSzGwdb-TGY9itsjV2I7BrP2LeR8IUz3fbfuzBauM5Lp5O248utiNVIkMaU8WMf3sPojDYC9d_tF15S3urVxJAWpMka0oXG9q1oYqujoRnql2XlAhiHgZ1bQcRUWSK5vhpH0X22I7s424EljjG5jgmcsGnLDVi7Yw2I6ooEMTLG27WE1IQjniggMfrw3g-9yEISENz4btoTiPulyg-sF13BWtvO_hVyhdR5DXAGlfAyDStKN3__AhjOdUKRraHhjG0UH_W3pskLaLJa3GKKzDkvy7uzl0Q4cwOZ78o6vL0eC1AMAWKyqX_f1tIkp3rrQ82_WURWUCS1x0EVWC4e_qDfSKr6m2AkGBTyxHCGlum4IfkiSNWRsZ2DmEpXbFWDHvGg50JfNAlTdbqsijLZl0yFE2uK6Kl4kNQYJA4AtXZ2SampzEfdcbQjo4_YUIsc-4GPfLht1kULcKT4-Plctlc2NHMn6AfjouaI39-vAAyIHaDaN_AfI5b8M_p9K7dejI7N-YUfvrwgRA8XlN8Tw9JQ0qe_USUwBKpkqwrEI8R9Z61zuTjBBb8ALb5E0YihmFGIgbLGUky_iIrXXdvlzf_s9L_rPTfaiVmGdli5qlrhizrddVUdPjGkEWlrkqqAZ_xeVOdVTFD1yz4XsSQsmFpdUmVTLmuWrKm1jVL18S6YoiGWZclSzPrqm5JYt20VCh9kiqLwEpTdeYbHchLdUlUJPxRsuogkKnAZ0U067qlWnVLluu6oqoGsLEUvS5hqSSQR6oriqQbdUU3MRdsKcAGPnXJ0hWQSjTgG1XUgLusW7oIemmghaybOkguWrIBssF_6ppkyUBTEkEsXVRJyIBl6goAgRaqrGGbGKCRpBga0NR0E6iB1mpdkQ2gr5iWCH8rqgEamKYIuMALsOALkBCwtLqmAF5dtrDdFMk0QEJFAeXgazCwgo0qy6qhAEMRTKgqgKlrRFtFFC2QTLSI8cBPpiICDtAAdVQVCBoawCmSAZYAl8jgH1UBxUQRf9QNgJAJe8lSDDCQBuxAIBmLq5i6Co40FWAHPIkJsPstC0ypaboKEoEtNew9BUwjqlgMkEkCl-ugKjjBwMayRKKYhEVVwHCKoRtSPuO0vyTjvkHGta8H16efGh9fPi-eb_Ru69vlVdjWT9uLVah8C0_n-s3HG8n9chH9VB2rvRrddXrLy-XNpdGIr8SGZV6s7s7mL5Mvn91pJ_wpTaZfG9eN587L97Nv_XDl97zv2uKuJ7ft648Txbjpr6J-cBUvvw7u0HnfNVovvi_Fj1fL4Gr2x4_FRJyG6Iv77ZsYvMhnsnV2M3q4Mqc_bz6effkxsqXJ1ZdJd9Cb6D_70eVz7LaW3Qtteedaz5d3lnk2F7917s_OHh_tiwf9tBd8X56Lp43O8OP4voG-hz3v7lm-j4Z3q9X84_Xi9tPo8vzmK-o-eD96ekO_vVq2Wv8plSdfc2gBOKjm0IRjNYckKK05yXJDS92m3NAgplmeKDe0ELBoTxQatsqQ8kELDS0xNGuSJYYWFJZTJL9oCaNlhWYfKyukoLCiVlJQWNXElbKkntBsp5mfqyesLGTrCSkctIbQekJKC60krASR-kRrVVkloYsErSS0ENOSysouKcrJSkJXBlpkaSWh1X5nJflr1u4vqUry7LQuBivLQtOb51HwSX5sfYuf2k-fvt3_CEaN7svgx1f3tD0NJ8vLxh-BM_70fPMjdFvawhwaV8vTXt_6IhktyMz-p6fVz4-rb4u7i--rO_ezeNYPbme3vW8Pyt3z9KspxRPPD66cL9bHzourPTyMPxrW1Ao7p1CZJt2-2J5KJtSuXuDG085QaYQ_e4PI_3bTb5_Go_Pu4O5C_DZpqeNb81OkB4ux6U6urltfbi7VQXch3YftxjL-OI3OB9ofE--76jfcvtS7WjUcf2Z_dr8HxqLv4YblX1Y13h_j7RX-8Ak2auQZ_nL9A9l83TtoKYTIDkazYRTgAvLT9-dfobhYalNTRdHA4akqdLB4ieyoby82m1gMy6aSGhtKRnZ05kA5CsjUdb2HxIBfPhw1IImbmmKJhmKqmkpvWbwnm1O2C8Sfh4huN-MQ0Us7D8he-B75OjHKxKDCix04Nt6eynSCyzbeAzKiOxHuPnVv7jq1vu3ao5nt_B_QCf8_FO4dHOO168BGkR8I7cBfejU62T4RJFNjn0_tcH1PDOIYntToXF2gg_UTQVaUWnLUnke4p_KdCNcBEpSm2BS3X7XjAKwU1eiRFxoLmydyRjQ2zT0RxBpz9Ilw2iB_asT-A_R8IuhmrTtfuM7Iwc_F2j8C25uiE7D4a2s8DoU3ktKQjLdC5AtvZKuhwEeySxfo2Lv2D2bME-k1gfrujSyKDVkTCeKcXpwSUobYYsppTEkDTHEnZhJckRuq-PZ3DI2PuAV6KIBPq2oZskCVwm2OqZPA5JoKCuwICZIopjkKCxRAwOOwEpyJYLuu0HmOncUC7M8CETzV9oMgXoBPkrQUTcAHMPtS2H7Cmdr3xwOsikAU-nAEGq0jl-VyKYRcCaFUQqiVEOurT2RmVJBoylqYokRru779JPgT4QxNHOKLnj16jqEajIULO3issTOhE8GUlfUPlWkGzczONBOblinr6VyTmkoy1_ZKPyUr9Sb_5EQCXjQ2fxJJqMnJJJSSgftGbegQtY43ChBoMK5OiTdQLxuQRCksZjXBhvgrSUPopdKJaIm49KbyEIfyFkPJJK5I-ZYjZFi8UcVGRk6SK5v0wRLjK2DJKvMwQ54wc6I6pOnvJGpIIRKcUIjwGZwwCXzKV3hEEx9ciE9Kk0IfjEfQ0hKiZG2o9R0wL76SAlAR9Nj_JWlMyRel7uc5Wvkj26u17cCzXeHU96Nwm6-SKpblK04_ScutjVbF0thD9jSGT11vFJOec2eaJVNMS69zklWUVCRc38lm-TKhiKlwZdfiBHKnoyb_LozIRQmMTi6ZCfSOGsk7Ov--BAFIzED3LvQTfOtCvMB4svh77asfCzP7BQmfvdC1Y5BDWM7ABNDTRQIkRc9fUll7fogEgwRmUgfCgPZBu8qyvA6MIt9euwgrAJQvQfqwdhXDj1TZyJ5M0mVSbuq8XUqWTqEDWaOys1mRk2VBAtu3N7Ynx_3JDkMy848pM1yu1-svO-aArx5X9LBFUETmt85o5qfLrAzlTk-XLwrKWiOoh6UxxpOoVW7TCt3Win46waZpPfedRxfVru3YFlrz2EXRwX5L0ti6RNJL-0foARvKvuX9nSaWrx_y6zvcm5WvL6nVxWxI8ttSDxS3n28kvSGrHFhSdiHT99M0zdWEUBK5CLBm0WNRS4qErP0uBPhSL6BtbhEBLjSYAcm08M8uP9xLxw4IvRLCKF9-9HSsD1qDTm0ApRmvsLDu0I-BPYGYPRHw9hE2Q9BN_jgRNma2TP21uzYvXtv1PIyiJ2CwByR5A4RfO6wN40n-h60nLet1c9mfuU1P4ySKmVrY9Elk84QTJxnnpBd5p1v574x8PugaRyAVeUJKxUORH4wCP6S3SrQFGPqwagewCLGOINObS2pVb26IaolDS3zYgz7E3TxLUmeUtQ3oetW-R7AOgB5asVvzP-xaqpKeNdI10VBTrXiq_X6nazmf7uwxiv1iVvrlwgU6Ccewn3O7JrPKM7ol_cs8Uw6q_2ofWWkfvVO08oHDvg7csXZrR-vPLTfCdY24zypcyh_wtUAUrNfyzgqb2gnnkMUj4QotkbtduVuQwMGJgN8iEMi91uJH18jLPVi_HSK0xmMCVQ7R9j3Y3UQZKHa57UQ4m0-b-S1UHo6AYG5pU-YBB9DC0u0UlLUSOvjdyO2zM-jogrnjkUp3wjq9sqdYLyy0QF-JKYWDx00BX8bGwKVQ97HrNYXufB57CaXPnHCE3-v2IHtoe5iVKQGQTlko1bBb4IEk5b0Y8Jyc9giD0SwA50DLHSQoXgT2CGDAFQ7WATxMXlnKAdC9DG2gcw8zBSr7uONNEIImNGeZSyzZ2ijFHBjIdhcEJvnouG4O4noGu0xvmvse_yaCJibreGGBAFGAixioD2lNpoesdc-BEOfnpWMPMWYqs9ZPcomDQRMdeIzjGG-xOi5t1FhM2uEekOlk3AVJdrS7QTsPOTNtwTAnKPPkLn4lWJV4FOo0DpoXQSUxfJVv3twB1sWhdh4g9BNVQVVZmAC1Z-5uuRhDKD6Pu62xGSNAqM8g8_lAK12WgIXgGjd5JXgqUuveidiqiSt2dgHJPM0Um8zTbNHKIZOKvtoCPEBXOjuhQmY5p56lMow9IWbCK0nGTsnH69yjIPmtrZZbVaF4CLA8b0FIm7PBTD3Saz1njo83oQvAbXzxblgV5UNGuWUYdDTJAbhr0yo1pKb29vd1kq0HosQFQIIEH3iMDp5scluATkZIV1CylWVEwFcowvNRfIKEx8LAhzDZjtKqCafmy_g9HYGc4jCZN7WVsQR5Oz-AZ0qZSh5qyiSyRg-TCsizKCJ-3o-F9rrZ2r3RYZ__Njl-3jF7LqDMOp8tbb3ARHJDMt-yWXViJLixT78o7HiYGenwYW7dTjzX3RobsKe4L_HYfuXH-IBWsAV6u5qDp5nk2cBMNbIxJr_RYdsAYn7sbRsOolZKEa1hptQg_VkBmVS7l0gqsYCave2osoG0ef16DxbSK54I41gR3ugijdH1xIc6lREngYSbFcJxGAXOEwr3YCS_3nlJONzTosB-dHAJ34OOwmFh0vgVCrdpXxMUM4lKJyTpEJ7a-I0qARJIQPZohp_OV3g14uWhpQO8CcVmXSGLkyZ7DkRH7rzsUsn7RmF-ndCGPcSdOunNMhxH5C1dXh5G-owu6wnS7rNDiiJXkB4-QS6VjutqVnzmwUPOKqwoZbOPaoKymInfzeajGlXaGbJUs9KQJduNBDGZJCwOyMTmBQemys4Kw_VWhoOY8ro5D2L7HA6kVL6kjYI3RWRsyvZF1cRSiUHvhWhmw6AzyjemCXGgvk3uoApp4j1RgqieOT5uwsLVNNlVjnQJW5_dcFA1OLuRzGURHtLmQb3Cnlaxis4wCnjQxoFtY8rp5_d4if5K3JfX7r5nJy9pX17b49fDGGaXMgYCnB6Q_eShMNyXovJK903Yq2wTvIsC2fhusdWi0vkrjJ3hk8122GETK-Jfsli4NpVS0jOUyCa8vBcopWMk7Ma26Luwu8myrJj7mo03H9JsrIyubcB08dlwtdVShNTsCrQdEfDTkBIGozODXbibRElQkPe12l7ZVsRRyRoQFjnc575UBUsRMTVDjBiBzxtF5LSyfS_T1a3c8q6nGAmieqZ3gXUF1oPEiISnLyyga2y6W0Jqn6a2gJr5uvtEPjWUISeDfHStMpPmAskJkyMFthnMciDzmsSWWeRcxYvu6HCQlw6bLHAkSZaTfFDPwM2o-DxdFeV_y4tlv-J2QOmptVlJwaqEkMRqkGqbStVGlaqtKlWbRDKqQarNIlXbRa62i8wRa9V2katjSa5WWq5WWq5WWqlWWqlWWuHIsOpgUKrtolTbReW46VOdqmq1ddVq62rVsmjVptOqTadxlKcdF1-hJBS_HeLBHtkP5rVzPwyhJabX_xx8TsgG-dmrFbJUcbm1_OpC8iKrVX5XfP2OQ8mFxOyNswzsjpcsNHG92U8fLKTHV3x3uR_Wk-A2GBCPgrGIdWFojwJn4sDyjV-OILjAjkxQ6Plsf3P2kV4bawj_Hgpyc1YjK-uGEhMhNXb5N7qIveuOCFtpCiPvu49nsrUL-yV5AYTe5hSz7y_ITTX_lZ5_y0HS-O6BaoxvYawOGvAneclGLLuvW5fryquauuc0jGJPOItZqMC_uNN1UGIDrtXVVylx15G2wgTvdhagcOZnjs8KgjzTdtXYoQGZhAnvJCFy5kmWRNC6WjioTTbR2XbNiWbrq9-UQ8FxUNnNz1yAv5NFEQ_CkDcFsol7nbV3CjzpehFkkDPFL8-lnibO_TQxOcxcZ976AvSWI60X7Aw3ccKn_Cl8sGBde70NnOkUNkOViHVqP1sQISzXb02tfxV_3jdl1EJhjMDRmWvYyYqIX8ViUlW9oqH8uh43kfdD11__Ekz6vqIgbV_UlDa_AXMLRW_tbWHMPMglcucoSpDJgzBecgJIzEPhS7VJGD0PcuqPV8L6Fxev4eQSUlIFKT9KKKbkIejd8y3IetFmr3_uePUz-ZZnkc2HS3shtB5XYYgnzvTtWW3LaNMHFqBm0WQ-tLyX88Y4UKIiZSqkOkVutCfKuWuHT4LCJZJcJJJymH35jCD_IjT1MCH1w7jph1tTqnBwUTxVeIDWkX0ZFUSSfgifivArqYMchJXDs0g6RBH1sFiQDos8tdo9PIbdl2xReB2oOFfd0Q6Q54AaqFSx2ax-e7qv2MoHxZd2CBJXSuZaiAJHHKSndrBZ5QP8rh-WScph4avsH2Xq4WX_gJiWDja-wpOZKg_QAWL_ieaH05P5ZpjbDfoBYXm4H_5EvPDWZIzLmTb8oAdYrmQLwlFj5IPNe2C94PdnbqfEoY7CYSdpt512uUE92FraIQsAT77ktoLcQV4hUtk28pjtI8kvQSKv2ZL_yYXvTZwpQLw_zv7vM_8JyelG9Q==
```

Reply:
From 
![image](https://user-images.githubusercontent.com/4302241/158328922-ef3472f0-9b2c-40ae-9319-c064c43f6357.png)

It selected the new one under it for me

To
![image](https://user-images.githubusercontent.com/4302241/158328664-a137a4f4-14c3-4f3e-ae92-e2d2b525434e.png)

Reply:
yep, there was a typo.